apps/src/codebridge/FilePreview/HTMLPreview.tsx

```diff
@@ -37,9 +43,21 @@ export const HTMLPreview: React.FC = () => {
     const re = /([-.]?studio)?\.?(cdn-)?code.org/i;
     const environmentKey = location.hostname.replace(re, '');
     const subdomain = environmentKey.length > 0 ? `${environmentKey}.` : '';
-    const port = 'localhost' === environmentKey ? `:${location.port}` : '';
-    return `${location.protocol}//preview.${subdomain}codeprojects.org${port}`;
-  }, []);
+    const useLocalPrefixOverride = experiments.isEnabledAllowingQueryString(
+      experiments.LOCAL_WEBLAB2_PREVIEW
+    );
+    const isLocalhost = 'localhost' === environmentKey;
+    // When testing on localhost, it can be convenient to have a fixed subdomain
+    // to avoid having to give permissions to every channel id version of the preview url.
+    // Use the flag ?local-weblab2-preview=true or ?enableExperiments=local-weblab2-preview
+    // to enable the fixed prefix locally.
+    const prefix =
+      useLocalPrefixOverride && isLocalhost
+        ? 'localtesting'
+        : normalizedChannelId;
```

Comment:
just confirming my understanding that this resolves  the TODO in `weblab2_project_service_worker.js`.

https://github.com/code-dot-org/code-dot-org/blob/d18ccc1806db8d411f9f7947bf15c88c7f4b6cae/apps/src/codebridge/FilePreview/weblab2_project_service_worker.js#L17-L20

Reply:
>If the same project is open in different tabs we will force reload a tab anyway, so I'm not too worried about that scenario. We generally don't support it.

Oh interesting - thanks!